src/context.c

```diff
@@ -130,6 +138,81 @@ static void set_ub_edns_maximum_udp_payload_size(struct getdns_context*,
 /* Stuff to make it compile pedantically */
 #define RETURN_IF_NULL(ptr, code) if(ptr == NULL) return code;
 
+
+#ifdef USE_WINSOCK
+// For windows, the CA trust store is not read by openssl. 
+// Add code to open the trust store using wincrypt API and add
+// the root certs into openssl trust store
+static int 
+add_WIN_cacerts_to_openssl_store(SSL_CTX* tls_ctx)
+{
+	HCERTSTORE      hSystemStore;
+	PCCERT_CONTEXT  pTargetCert = NULL;
+
+	// load just once per context lifetime for this version of getdns
+	// TODO: dynamically update CA trust changes as they are available
+	if (!tls_ctx)
+		return 0;
+
+	// Call wincrypt's CertOpenStore to open the CA root store.
+
+	if ((hSystemStore = CertOpenStore(
+		CERT_STORE_PROV_SYSTEM,
+		0,
+		NULL,
+		// NOTE: mingw does not have this const: replace with 1 << 16 from code
+		// CERT_SYSTEM_STORE_CURRENT_USER,
+		1 << 16,
+		L"root")) == 0)
+	{
+		return 0;
+	}
+
+	X509_STORE* store = SSL_CTX_get_cert_store(tls_ctx);
+	if (!store)
+		return 0;
+
+	// iterate over the windows cert store and add to openssl store
+	while (pTargetCert = CertEnumCertificatesInStore(
+		hSystemStore,
+		pTargetCert))
+	{
+		X509 *cert1 = d2i_X509(NULL, 
+			(const unsigned char **)&pTargetCert->pbCertEncoded,
+			pTargetCert->cbCertEncoded);
+		if (!cert1) {
+			// do not return if a cert fails, continue and retrieve the rest
+			DEBUG_STUB("*** %s(%s %d:%s)\n", __FUNCTION__,
+				"unable to parse certificate in memory",
+				ERR_get_error(), ERR_error_string(ERR_get_error(), NULL));
+			return 0;
+		}
+		else {
+			// do not return if a cert add to store fails, continue and retrieve the rest
+			if (X509_STORE_add_cert(store, cert1) == 0) {
+				DEBUG_STUB("*** %s(%s %d:%s)\n", __FUNCTION__,
+					"error adding certificate", ERR_get_error(),
+					ERR_error_string(ERR_get_error(), NULL));
+				return 0;
+			}
+			X509_free(cert1);
+		}
+	}
+
```

Comment:
Last comment (I hope!). 
1) Could you update the code comments above to reflect the updated logic
2) I believe that if the cert store is empty this function will return success. Again, I think we should fail in this case since then Authentication will not be possible.